cosmwasm/cli-utils.js

```diff
@@ -112,7 +112,31 @@ const addContractOptions = (program) => {
 };
 
 const addStoreOptions = (program) => {
-    program.addOption(new Option('-a, --artifactPath <artifactPath>', 'artifact path').env('ARTIFACT_PATH'));
+    program.addOption(
+        new Option('-a, --artifactPath <artifactPath>', 'Path to the artifact file to upload (required if --version is not used)')
+            .env('ARTIFACT_PATH')
+            .conflicts('version'),
+    );
+
+    program.addOption(
+        new Option(
+            '-v, --version <contractVersion>',
+            'Specify a released version (vX.Y.Z) or a commit hash to upload (required if --artifactPath is not used)',
+        )
+            .env('CONTRACT_VERSION')
+            .conflicts('artifactPath'),
+    );
+
+    program.hook('preAction', async (thisCommand) => {
+        const opts = thisCommand.opts();
+
+        if (!opts.artifactPath && !opts.version) {
+            throw new Error('Either --artifactPath or --version is required');
+        }
+
+        const wasmResolvedPath = await getWasmFilePath(opts, opts.contractName);
+        Object.assign(opts, { wasmResolvedPath });
```

Comment:
this helper doesn't know that `contractName` is a valid option, so it's adding a weird coupling. I'd rather just have the actual script call `getWasmFilePath` instead of doing funky things with the options during the CLI setup

Reply:
hmm, it is convenient to remove duplication. I'd prefer that contractName be defined as an option in the same function then, but might be a bit trickier. But we can try this approach out